### PR TITLE
BartekK/HOL-31-handling-vtk-format

### PR DESCRIPTION
**Pull request by tpxkuchnowski**
This PR allows loading VTK models (lines and meshes). Suitable document concerning vtk file structure will be provided

### CODE REVIEW

File: unity/Holo/Assets/Editor/ModelPreprocessing/AssetBundles/AssetBundleCreator.cs [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
+        SaveFilesForExport();
+        BuildMapABs();
+
+        BuildPipeline.BuildAssetBundles(Application.dataPath + "/StreamingAssets", buildMapArray, BuildAssetBundleOptions.None, BuildTarget.StandaloneWindows);
+    }
+
+    // Exports finished GameObject to a .prefab
+    private void SaveFilesForExport()
+    {
+        if (!AssetDatabase.IsValidFolder(@"Assets\" + ModelGameObject.name))
```
Comment (ellendil) [inline unity/Holo/Assets/Editor/ModelPreprocessing/AssetBundles/AssetBundleCreator.cs]: Please use the surrounding `{ ... }` for the `if` `else` and other code blocks

File: unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs [hunk 2]
```diff
@@ -1,91 +1,147 @@
...
         GetGameObjectName(rootDirectory);
         LoadFiles();
+        ConfigureMesh();
     }
 
+    //Loads file paths of particular frames and their extension
     private void GetFiles(string rootDirectory)
     {
         filePaths = Directory.GetFiles(rootDirectory + @"\");
         extension = Path.GetExtension(filePaths[0]);
```
Comment (ellendil) [inline unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs:33]: We should use first known extentions not first from all files.
Please log also information if there are more than 1 extension in the directory.

File: unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/VTKImport/PolyDataImporter.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public Vector3[] Normals { get; private set; }
+
+    public bool facetsInwards = false;
+    public void LoadFile(StreamReader streamReader)
+    {
+        GetVertices(streamReader);
+        SkipLines(streamReader);
+        GetIndicesAndNormals(streamReader);
+    }
+
```
Comment (ellendil) [inline unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/VTKImport/PolyDataImporter.cs:21]: Double empty line

**Review by ellendil — APPROVED** [review]
LGTM

File: unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs (hunk 2 above)
Comment (tpxkuchnowski) [reply unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/FileSeriesImporter.cs:33]: I'm assuming that in the folder are only files with the needed meshes. I've added consistency check, that ensures that every file is the same format as the first one. I'll add it to import requirements along with VTK format specification.

File: unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/VTKImport/PolyDataImporter.cs (hunk 3 above)
Comment (tpxkuchnowski) [reply unity/Holo/Assets/Editor/ModelPreprocessing/ModelImport/VTKImport/PolyDataImporter.cs:21]: Fixed.

File: unity/Holo/Assets/Editor/ModelPreprocessing/AssetBundles/AssetBundleCreator.cs (hunk 1 above)
Comment (tpxkuchnowski) [reply unity/Holo/Assets/Editor/ModelPreprocessing/AssetBundles/AssetBundleCreator.cs]: Done.

**Pull request merged**